pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/Consumer.java

```diff
@@ -185,8 +186,10 @@ public String consumerName() {
                     readChecksum(metadataAndPayload);
                 }
 
+                MessageMetadata messageMetaData = Commands.parseMessageMetadata(metadataAndPayload);
```

Comment:
I think we don't have to deserialize messageMetadata again. at line #160 we already know number of messages which is going to be dispatched. can we use that number?